src/Encoder.hack

```diff
@@ -117,9 +117,149 @@ class HTMLPurifier_Encoder {
             return $str;
         }
 
-        // If anything does not match the regex, it is the 1% that doesn't and thus we really do NOT 
-        // want to be supporting that, so we just throw an exception instead
-        throw new \Exception('This string is in the 1% that does not fall in the main case');
+		$mState = 0; // cached expected number of octets after the current octet
```

Comment:
Is the indentation funky here or is that just GitHub? Looks like tabs vs spaces?

Reply:
Yes, there is an indentation problem.